src/couch/test/eunit/couch_key_tree_tests.erl

```diff
@@ -561,10 +561,15 @@ should_not_use_excessive_memory_when_stemming() ->
         % Without the optimization #91de482fd66f4773b3b8583039c6bcaf1c5727ec
         % stemming would consume about 18_000_000 words. With it, it consumes
         % 6_000_000. So, use 13_000_000 as a threshold.
+        Size =
+            case ?COUCHDB_JS_ENGINE of
+                "quickjs" -> 18_000_000;
+                "spidermonkey" -> 21_000_000
```

Comment:
Oh interesting, does it seem like it js_engine related? That is a bit unexpected as we're just generating a plain re-tree. Is it fairly consistent?

Reply:
Yes, quickjs is always smaller than spidermonkey 1.8.5.
- If I run test using `make eunit apps=couch suites=couch_key_tree_tests tests=key_tree_stemming_test_`, then `quickjs -> 18_000_000, spidermonkey -> 21_000_000`;
- If I run `make eunit`, they use more heap, `quickjs -> 19_000_000, spidermonkey -> 22_000_000`.

Reply:
https://github.com/apache/couchdb/pull/5190

Reply:
Merged. Thank you for the review.